webpack.config.js

```diff
@@ -1,31 +1,47 @@
-let path = require('path');
+const path = require('path');
 
-module.exports = env => {
-    console.log('env: ', env);
+const config = {
+  entry: path.resolve(__dirname, 'src/client/index'),
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    publicPath: '/',
+    filename: 'bundle.js'
+  },
+  devServer: {
+    contentBase: path.resolve(__dirname, 'public'),
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: 'babel-loader'
+      },
+      {
+        test: /\.s?css$/,
+        use: ['style-loader', 'css-loader', 'sass-loader'],
+      },
+      {
+        test: /\.(png|jpg|gif)$/,
+        use: [
+          {
+            loader: 'file-loader',
+            options: {}
+          }
+        ]
+      }
+    ]
+  }
+};
+
+module.exports = (env, argv) => {
+
+  if (argv.mode === 'production') {
+    config.mode = 'production';
+  }
+  if (argv.mode === 'development') {
+    config.mode = 'development';
+  }
```

Comment:
@loginov-rocks is that what you meant with --mode options?

Reply:
Webpack consumes mode argument by itself, so this is not needed.